From 4e7d43fbd2fabb32ff268a749de84d24527fb222 Mon Sep 17 00:00:00 2001
From: Markus Valentin <valy@systemausfall.org>
Date: Tue, 14 Mar 2017 16:05:02 +0100
Subject: [PATCH 4/4] spi: spidev: add a compatible string to spidev driver

Signed-off-by: Markus Valentin <valy@systemausfall.org>
---
 drivers/spi/spidev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spidev.c b/drivers/spi/spidev.c
index d0e7dfc647cf..57b0760cf002 100644
--- a/drivers/spi/spidev.c
+++ b/drivers/spi/spidev.c
@@ -695,6 +695,7 @@ static struct class *spidev_class;
 static const struct of_device_id spidev_dt_ids[] = {
 	{ .compatible = "rohm,dh2228fv" },
 	{ .compatible = "lineartechnology,ltc2488" },
+	{ .compatible = "spidev" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, spidev_dt_ids);
-- 
2.11.0

