From c299aa85342020969409ce8d8f980847d1363aca Mon Sep 17 00:00:00 2001
From: Konrad Mohrfeldt <mohrfeldt@silicann.com>
Date: Fri, 23 Jun 2017 18:07:09 +0000
Subject: [PATCH 8/8] change duart muxing

---
 board/silicann/urwerk/iomux.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/board/silicann/urwerk/iomux.c b/board/silicann/urwerk/iomux.c
index ef9f8af265..eaf46cb4ae 100644
--- a/board/silicann/urwerk/iomux.c
+++ b/board/silicann/urwerk/iomux.c
@@ -23,12 +23,12 @@
 
 const iomux_cfg_t iomux_setup[] = {
 	/* DUART  ON INTERNAL TESTPADS*/
-	MX28_PAD_PWM0__DUART_RX,
-	MX28_PAD_PWM1__DUART_TX,
+	/*MX28_PAD_PWM0__DUART_RX,*/
+	/*MX28_PAD_PWM1__DUART_TX,*/
 
  	/* DUART ON EXTERNAL UART */
-	/*MX28_PAD_AUART0_CTS__DUART_RX,*/
-	/*MX28_PAD_AUART0_RTS__DUART_TX,*/
+	MX28_PAD_AUART0_CTS__DUART_RX,
+	MX28_PAD_AUART0_RTS__DUART_TX,
 
     /*UART0*/
     MX28_PAD_AUART0_RX__AUART0_RX,
-- 
2.11.0

